deepinterpolation/generator_collection.py

```diff
@@ -536,126 +398,163 @@ def __data_generation__(self, local_x, local_y, local_z, local_t):
         return input_full, output_full
 
 
-class EphysGenerator(DeepGenerator):
-    "Generates data for Keras"
+class SequentialGenerator(DeepGenerator):
+    """This generator stores shared code across generators that have a
+    continous temporal direction upon which start_frame, end_frame,
+    pre_frame,... are used to to generate a list of samples. It is an abstract
+    class that is meant to be extended with details of how datasets
+    are loaded."""
 
     def __init__(self, json_path):
         "Initialization"
         super().__init__(json_path)
 
-        self.raw_data_file = self.json_data["train_path"]
-        self.batch_size = self.json_data["batch_size"]
-
+        # We first store the relevant parameters
         if "pre_post_frame" in self.json_data.keys():
             self.pre_frame = self.json_data["pre_post_frame"]
             self.post_frame = self.json_data["pre_post_frame"]
         else:
             self.pre_frame = self.json_data["pre_frame"]
             self.post_frame = self.json_data["post_frame"]
 
-        self.pre_post_omission = self.json_data["pre_post_omission"]
-        self.start_frame = self.json_data["start_frame"]
-        self.steps_per_epoch = self.json_data["steps_per_epoch"]
-
-        # This is used to limit the total number of samples
-        # -1 means to take all and is the default fall back
-
         if "total_samples" in self.json_data.keys():
             self.total_samples = self.json_data["total_samples"]
         else:
             self.total_samples = -1
 
-        # This is compatible with negative frames
+        if "randomize" in self.json_data.keys():
+            self.randomize = self.json_data["randomize"]
+        else:
+            self.randomize = 1
+
+        if "pre_post_omission" in self.json_data.keys():
+            self.pre_post_omission = self.json_data["pre_post_omission"]
+        else:
+            self.pre_post_omission = 0
+
+        # load parameters that are related to training jobs
+        self.batch_size = self.json_data["batch_size"]
+        self.steps_per_epoch = self.json_data["steps_per_epoch"]
+
+        # Loading limit parameters
+        self.start_frame = self.json_data["start_frame"]
         self.end_frame = self.json_data["end_frame"]
 
-        self.nb_probes = 384
+        # start_frame starts at 0
+        # end_frame is compatible with negative frames. -1 is the last
+        # frame.
 
-        self.raw_data = np.memmap(self.raw_data_file, dtype="int16")
+        # We initialize the epoch counter
+        self.epoch_index = 0
+
+    def update_end_frame(self, total_frame_per_movie):
```

Comment:
Looks like this method might be private. If so, it should begin with `_`

Reply:
True, updated